src/clientdlg.cpp

```diff
@@ -1190,16 +1191,18 @@ void CClientDlg::OnCLPingTimeWithNumClientsReceived ( CHostAddress InetAddr, int
 
 void CClientDlg::Connect ( const QString& strSelectedAddress, const QString& strMixerBoardLabel )
 {
+    // ---> pgScorpio: This code does not belong here but in CClient !!
+
     // set address and check if address is valid
     if ( pClient->SetServerAddr ( strSelectedAddress ) )
     {
         // try to start client, if error occurred, do not go in
         // running state but show error message
         try
         {
-            if ( !pClient->IsRunning() )
+            if ( !pClient->SoundIsStarted() ) // ---> pgScorpio: Again this is NOT connection started !() )
```

Comment:
No, it's whether the client is running.

Reply:
Exactly what I mean to say! This function does NOT indicate if we are connected to a server, but on;ly if Sound is Started.
And Sound started does say nothing about an connection to the server, since Sound.Start() can have failed (No check on that) but the connection still can be started with Channel.Enable(true) !
This is exactly where it goes wrong with an invalid device selected...
Sound is NOT started, but the Channel is enabled !

Reply:
It's meant to indicate whether the client has _started_.  Not the state of the sound system.  Not the state of the network connection.  But whether the client has started.

Reply:
> It's meant to indicate whether the client has _started_. Not the state of the sound system. Not the state of the network connection. But whether the client has started.

I Think we mean the same, but use different vocabulary ;=))
And yes, I am very persistent in using variable- and function names that clearly state what they do (And unfortunately I encounter a lot of them in jamulus which absolutely don't, some of them even very misleading.)

So I think the Client is already **started** when you run jamulus (without a -s ;=))

It's about if the connection to the server is started, and the original IsRunning() (Used at many places) had nothing to do with the connection to the server, but with Sound. So if Sound.Start() fails (and there is no check on that!) they all think there is no connection to the server, but in reality we actually are connected/connecting to the server, so UI and reality don't match and you can't disconnect, since hitting the connect button will try to start a connection again while we are already connecting.

Reply:
No, the Client is definitely _not_ started when you run Jamulus.  It's not even created then.

Reply:
> No, the Client is definitely _not_ started when you run Jamulus. It's not even created then.

I meant It is started automatically when you run jamulus  (without a -s ;=)). (Created in main and started with `pApp->exec();`)

As said we are apparently using different vocabulary...
Maybe you call " Started" what I call "Connected" (or "connecting" ?), since I think the client is already started if you see it's user interface and can interact with it. Though it's just not yet connected to a server, it's still "Running" in some way. 

Simple verbs like " Start" and " Stop" are often ambiguous, and adverbs like " Running"  and " Connected" are often even more ambiguous and often interpreted differently depending on the context. 
i.e. When are you "Driving" a car ?, As soon as you take a seat behind the wheel? (Legally seen yes), When you "Started" the engine? (You can drive downhill without even starting the engine), or when your car is really moving ? (If so, also when it's on the back of a truck?, or being towed?) 

So to come back on topic: " Started" does not mean "Running", nor "Connected". And **not "Connected"** is not the same as **not "Running"** (might be "Connecting", so " Started"  but not yet " Connected" ). An easy confusion that can cause a lot of, often hard to find, bugs, like in this case.

But no offence, I'm the last one to say that writing clear code is easy. Even after so many years I'm still struggling almost daily with good, clear, naming's.

Reply:
Don’t quote me on that – since my knowledge of the code is basic – but I think "the client" is not the UI but the CClient class. Maybe talking about the actual class names makes it easier to avoid confusion. 

Also: I wouldn’t write your name next to comments since they might be updated/… in future by someone else who isn’t you. Having you in the git history should clearly show who wrote what if this is actually needed.